app/controllers/api/v1/PostsController.js

```diff
@@ -90,6 +90,7 @@ exports.addController = function(app) {
       return res.status(401).jsonp({ err: 'Not found' })
 
     models.Post.getById(req.params.postId)
+      .then(function(post) { return post.validateLikeOrUnlike('like', req.user.id) })
```

Comment:
Validation should never be a part of controller, let's push it a little further and extract to User model. The errors you saw in your last pull request #159 are OK. You may just want to update unit/functional tests to cover this.